src/test/java/com/xwiki/authentication/saml/samlauth/SamlAuthenticatorTest.java

```diff
@@ -0,0 +1,678 @@
+
+            return this;
+        }
+
+        public GivenDSL xwiki(ConsumerWithThrowable<XWikiUsersDSL,Exception> configuration) throws Exception {
+            configuration.accept(new XWikiUsersDSL());
+            return this;
+        }
+
+        public GivenDSL userIsAnonymous() {
+            when(samlAuth.isAuthenticated()).thenReturn(false);
+            when(samlAuth.getNameId()).thenReturn(null);
+            return this;
+        }
+
+        public GivenDSL userIsLoggedInByCookie(String loggedUserName){
+            loggedWikiUser = mock(XWikiUser.class);
+            when(loggedWikiUser.getFullName()).thenReturn(loggedUserName);
+            when(request.getCookie("username")).thenReturn(mock(Cookie.class));
+            return this;
+        }
+
+        public GivenDSL userIsLoggedInTheSession(String loggedUserName){
+             when(context.getRequest().getSession(true).getAttribute(any())).thenReturn(loggedUserName);
+            return this;
+        }
+
+        public GivenDSL currentRequestUrlIs(String currentRequestUrl){
+            when(request.getRequestURL()).thenReturn(new StringBuffer(currentRequestUrl));
+            return this;
+        }
+
+        public GivenDSL identityProviderAuthenticatedUser(Consumer<IdpUserData> userConfiguration) {
+            when(request.getParameter("SAMLResponse")).thenReturn("Some SAML Response");
+            when(samlAuth.isAuthenticated()).thenReturn(true);
+            IdpUserData idpUserData = new IdpUserData();
+            userConfiguration.accept(idpUserData);
+            when(samlAuth.getNameId()).thenReturn(idpUserData.id);
+
+            final Map<String, List<String>> samlAttributes = new LinkedHashMap<>();
+            samlAttributes.put("firstName", singletonList(idpUserData.firstName));
+            samlAttributes.put("lastName", singletonList(idpUserData.lastName));
+            samlAttributes.put("email", singletonList(idpUserData.id));
+            samlAttributes.put("XWikiGroups", singletonList(idpUserData.newGroup));
+
+            when(samlAuth.getAttributesName()).thenReturn(new ArrayList<>(samlAttributes.keySet()));
+            when(samlAuth.getAttributes()).thenReturn(samlAttributes);
+
+            return GivenDSL.this;
+        }
+
+        public GivenDSL shouldCapitalize(boolean shouldCapitalize) {
+            props.setProperty("xwiki.authentication.saml2.xwiki_user_rule_capitalize", shouldCapitalize+"");
+            return this;
+        }
+
+        public ThenDSLForDefaultHandler whenUserAccess() {
+            final AtomicBoolean handled = new AtomicBoolean(false);
+            try {
+                subject.checkAuth(context, () -> {
+                        handled.set(true);
+                        return null;
+                    }
+                );
+                return new ThenDSLForDefaultHandler(handled.get());
+            }catch (XWikiException e){
```

Comment:
```suggestion
            } catch (XWikiException e) {
```

Make your IDE do these. There are quite a few inconsistencies with whitespace and braces around the place, the three methods above for example.